build.gradle.kts

```diff
@@ -117,7 +117,10 @@ tasks.withType<Test>().configureEach {
     }
 }
 
-val supportedVersions = file("src/main/resources/versions.json").readText().fromJson<Versions>()?.supportedVersions
+@Suppress("UNCHECKED_CAST")
+val supportedVersions = (JsonSlurper()
+    .parse(file("src/main/resources/versions.json")) as Map<String, Map<String, Array<String>>>)["supportedVersions"]
+
```

Comment:
thanks @erichaagdev for all the suggestions. 
I applied all the changes here https://github.com/gradle/android-cache-fix-gradle-plugin/pull/498/commits/cb17b5a5e49b53ecb994fd87bca5c6437a816cb2

Because it was updated the upload logic I retest again the action of publishing a plugin:
https://plugins.grdev.net/plugin/org.gradle.android.cache-fix